doc/Requirements.md

```diff
@@ -0,0 +1,184 @@
+More specifically, if the identifier is a string it should be formated the same
+way as a file system path. Eg the parameter named "Root/child1/4/parameter1".
+(Why: Usual syntax to address trees.)
+
+
+# Syncer
+
+The PF philosophy is to map the hardware characteristics to parameters.
+
+A syncer MUST be mapped to one or more parameters.
+(Why: The hardware minimal access may be bigger than one parameter.)
+
+A syncer MUST support retrieving the mapped parameters value from the mapped hardware.
+(Why: to retrieve a parameter value at the start of the PF.)
+
+A syncer MUST support setting the mapped parameters value to the mapped hardware.
+(Why: to synchronise hardware on parameter change.)
+
+Syncers MUST retrieve and set the parameters value from the PF core.
+(Why: to be able to synchronise with hardware.)
+
+This API MAY be a packed parameter structure, following the C abi without padding.
+(Note: This is what the reference implementation does.)
+(Why: ???)
+
+## Introspection
+The syncer API SHOULD allow introspection of the mapped parameters.
+(Why: the parameter structure may be useful for the syncer to communicate with the hardware.)
+
+## Plugins
+The PF MUST be able to create syncers.
+(Why: to bind on the corresponding parameters.)
+
+### Definition
+The PF creates syncer using syncer builder.
+
+### Identifier
+All syncer mapping the same hardware SHOULD have their builder regrouped in a syncer library.
+(Why: to be able to link a group of parameters and a given hardware.)
+
+A syncer builder MUST have a unique identifier in it's containing syncer library.
+(Why: To uniquely identify the syncer that should bind on parameters.
+      Given that the syncer library has already been specified.)
+
+A syncer library MUST have a unique identifier in the host system.
+(Why: To identify the library associated to parameters.)
+
+Syncer build and syncer library identifiers SHOULD be strings.
+
+Syncer are dedicated to map one hardware. Multiple syncers that can talk
+Plugins contain subsystem libraries.
+A subsystem library MUST
+
+## Mapping
+One or more
+
+# Domains
+
+## Selection criterion
+
+A selection criterion MUST have one, and only one, state at a given time.
+
+A selection criterion SHOULD know at construction all possible states.
+(Why: To be able to validate:
+       - rules on start
+       - state changes)
+
+The selection criterion possible states SHOULD be specifiable by at least:
+ - Directly a state set (Input -> states == identity)
+   (Note: called exclusive criterion)
+ - A combination of values
+   (Note: combination in the mathematical sense “ab” -> [“”, “a”, “b”, “ab”])
+   (Note: called inclusive criterion)
+
+Parameter values MUST be able to change based on selection criterion state.
+Note: this does not mean that a function can be defined between both states.
+      Ie: Parameter values MAY depend on previous selection criterion states.
```

Comment:
It does ! Maybe my sentence was not clear. If a domain has no applicable configuration, it's parameters will keep the their old values.